lib/que/metrics.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+require "rack"
+require "benchmark"
+require "prometheus/client"
+require "prometheus/middleware/exporter"
+
+module Que
+  class Metrics
+    def initialize(labels: {})
+      @registry = Prometheus::Client::Registry.new
+      @base_labels = labels
+
+      @jobs_worked_total = register_counter(
+        :que_jobs_worked_total, "Counter for all jobs processed"
+      )
+      @jobs_worked_seconds_total = register_counter(
+        :que_jobs_worked_seconds_total, "Sum of the time taken processing each job class"
+      )
+      @jobs_latency_seconds_total = register_counter(
+        :que_jobs_latency_seconds_total, "Sum of time spent by job and priority waiting in queue"
+      )
+      @worker_running_seconds_total = register_counter(
+        :que_worker_running_seconds_total, "Time since starting to work jobs"
+      )
+      @worker_job_acquire_total = register_counter(
+        :que_worker_job_acquire_total, "Counter of number of attempts to acquire jobs"
+      )
+      @worker_job_acquire_seconds_total = register_counter(
+        :que_worker_job_acquire_seconds_total, "Sum of time taken to acquire jobs"
+      )
+    end
+
+    def expose(port: 8080)
+      Rack::Handler::WEBrick.run(
```

Comment:
Did you have a chance to think about the logging? 

With these changes the stdout of the worker will have the following as well:
```
Starting worker...
[2018-04-18 11:18:38] INFO  WEBrick 1.4.2
[2018-04-18 11:18:38] INFO  ruby 2.5.1 (2018-03-29) [x86_64-darwin16]
[2018-04-18 11:18:38] INFO  WEBrick::HTTPServer#start: pid=99417 port=8080
::1 - - [18/Apr/2018:11:18:41 BST] "GET /metrics HTTP/1.1" 200 1012
- -> /metrics
::1 - - [18/Apr/2018:11:18:51 BST] "GET / HTTP/1.1" 200 7
- -> /
::1 - - [18/Apr/2018:11:18:52 BST] "GET /favicon.ico HTTP/1.1" 200 7
http://localhost:8080/ -> /favicon.ico
::1 - - [18/Apr/2018:11:18:56 BST] "GET /metrics HTTP/1.1" 200 1012
- -> /metrics
::1 - - [18/Apr/2018:11:18:58 BST] "GET /metrics HTTP/1.1" 200 1011
- -> /metrics
::1 - - [18/Apr/2018:11:19:11 BST] "GET /metrics HTTP/1.1" 200 1011
```

Reply:
Pushed a commit that prevents these logs. Think you're right and it doesn't make much sense to include them.